var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -46,20 +46,54 @@ class Rose(Package):
     depends_on("autoconf@2.69", type='build')
     depends_on("automake@1.14", type='build')
     depends_on("libtool@2.4", type='build')
-    depends_on("boost@1.54.0")
-    depends_on("jdk@8u25")
+    depends_on("boost@1.47.0:")
 
-    def install(self, spec, prefix):
-        # Bootstrap with autotools
+    variant('tests', default=False, description='Build the tests directory')
+
+    variant('binanalysis', default=False, description='Enable binary analysis tooling')
+    depends_on('libgcrypt', when='+binanalysis', type='build')
+    depends_on('py-binwalk', when='+binanalysis', type='run')
+
+    variant('c', default=True, description='Enable c language support')
+    variant('cxx', default=True, description='Enable c++ language support')
+
+    variant('fortran', default=False, description='Enable fortran language support')
+
+    variant('java', default=False, description='Enable java language support')
+    depends_on('jdk', when='+java')
+
+    variant('z3', default=False, description='Enable z3 theorem prover')
+    depends_on('z3', when='+z3')
+
+    build_directory = 'spack-build'
+
+    def autoreconf(self, spec, prefix):
         bash = which('bash')
         bash('build')
 
-        # Configure, compile & install
-        with working_dir('rose-build', create=True):
-            boost = spec['boost']
+    @property
+    def languages(self):
+        spec = self.spec
+        langs = [
+            'binaries' if '+binanalysis' in spec else '',
+            'c' if '+c' in spec else '',
+            'c++' if '+cxx' in spec else '',
+            'java' if '+java' in spec else '',
+            'fortran' if '+fortran' in spec else ''
+        ]
+        return list(filter(None, langs))
+
+    def configure_args(self):
+        spec = self.spec
+        return [
+            '--disable-boost-version-check',
+            "--with-alternate_backend_C_compiler={0}".format(self.compiler.cc),
+            "--with-alternate_backend_Cxx_compiler={0}".format(self.compiler.cxx),
```

Comment:
I don't think so. This should point to the compiler rose will be using as a compiler backend, not necessarily the compiler used to compile rose.